lib/pbench/server/api/resources/upload_api.py

```diff
@@ -67,23 +96,61 @@ def __init__(self, config, logger):
         self.temporary.mkdir(mode=0o755, parents=True, exist_ok=True)
         self.logger.info("Configured PUT temporary directory as {}", self.temporary)
 
-    @Auth.token_auth.login_required()
-    def put(self, filename: str):
-        disabled = ServerConfig.get_disabled()
-        if disabled:
-            abort(HTTPStatus.SERVICE_UNAVAILABLE, **disabled)
+    def _put(self, args: ApiParams, request: Request, context: ApiContext) -> Response:
+        """Upload a dataset to the server.
+
+        The client must present an authentication bearer token for a registered
+        Pbench Server user.
+
+        We get the requested filename from the URI: /api/v1/upload/<filename>.
+
+        We get the originating controller nodename from a custom "controller"
+        HTTP header.
+
+        We get the dataset's resource ID (which is the tarball's MD5 checksum)
+        from the "content-md5" HTTP header.
```

Comment:
Is it worth mentioning that if the computed MD5 of the data stream does not match the `Content-MD5' header (do the capitals matter?), a 400 is returned?